Websockets/feeds/bybit.py

```diff
@@ -0,0 +1,342 @@
+                    ])
+
+    def get_topic_symbol(self, message_topic):
+        # +1 for the "."
+        return self.get_pair_from_exchange(message_topic[message_topic.rfind(".") + 1:])
+
+    def get_topic_time_frame(self, topic_name, message_topic):
+        return self.interval_to_time_frame(re.search(f"{topic_name}\.(.+?)\.", message_topic).group(1))
+
+    def _parse_side(self, side):
+        return BUY if side == BybitSide.BUY.value else SELL
+
+    def fix_precision(self, price, exp):
+        return price * 10 ** -exp if price else None
+
+    async def _send_command(self, command, args=None):
+        await self.websocket.send(json.dumps({"op": command, "args": args or []}))
+
+    async def ping(self):
+        await self._send_command(command="ping")
+
+    async def do_auth(self):
+        expires = self.generate_nonce()
+        signature = self.generate_signature(expires)
+        await self._send_command(command="auth", args=[self.api_key, expires, signature])
+
+    @classmethod
+    def get_name(cls):
+        return 'bybit'
+
+    @classmethod
+    def get_ws_endpoint(cls):
+        return 'wss://stream.bybit.com/realtime'
+
+    @classmethod
+    def get_ws_testnet_endpoint(cls):
+        return 'wss://stream-testnet.bybit.com/realtime'
+
+    @classmethod
+    def get_endpoint(cls):
+        return 'https://api.bybit.com'
+
+    @classmethod
+    def get_testnet_endpoint(cls):
+        return 'https://api-testnet.bybit.com'
+
+    @classmethod
+    def get_ccxt_async_client(cls):
+        return bybit
+
+    @classmethod
+    def get_L2_book_feed(cls):
+        return BybitFeeds.L2_BOOK.value
+
+    @classmethod
+    def get_L3_book_feed(cls):
+        return Feeds.UNSUPPORTED.value  #  'orderBookL2'
+
+    @classmethod
+    def get_trades_feed(cls):
+        return BybitFeeds.TRADES.value
+
+    @classmethod
+    def get_ticker_feed(cls):
+        return BybitFeeds.TICKER.value
+
+    @classmethod
+    def get_candle_feed(cls):
+        return Feeds.KLINE.value
+
+    @classmethod
+    def get_kline_feed(cls):
+        return BybitFeeds.KLINE.value
+
+    @classmethod
+    def get_funding_feed(cls):
+        return Feeds.TICKER.value
+
+    @classmethod
+    def get_mark_price_feed(cls):
+        return Feeds.TICKER.value
+
+    @classmethod
+    def get_margin_feed(cls):
+        return Feeds.UNSUPPORTED.value
+
+    @classmethod
+    def get_position_feed(cls):
+        return Feeds.POSITION.value
+
+    @classmethod
+    def get_portfolio_feed(cls):
+        return Feeds.UNSUPPORTED.value
+
+    @classmethod
+    def get_orders_feed(cls):
+        return BybitFeeds.ORDERS.value
+
+    @classmethod
+    def get_execution_feed(cls):
+        return BybitFeeds.TRADE.value
+
+    """
+    Authentication
+    from https://github.com/bybit-exchange/api-connectors/blob/master/official-ws/python/BybitWebsocket.py
+    """
+
+    def generate_nonce(self):
+        return str(int(round(time.time()) + 1)) + "000"
```

Comment:
It might be faster to do:
```python
return str(int(round(time.time()) + 1)*1000)
```
Since this is an int number, there should not be any precision issue